File: packages/protocol/src/capabilities.ts
export type Capability =
  | 'input.mouse'
  | 'input.keyboard'
  | 'media.control'
  | 'presentation.control'
  | 'clipboard.read'
  | 'clipboard.write'
  | 'files.read'
  | 'files.write'
  | 'apps.launch'
  | 'windows.control'
  | 'automation.execute'
  | 'power.lock'
  | 'power.sleep'
  | 'power.restart'
  | 'power.shutdown';

export const ALL_CAPABILITIES: Capability[] = [
  'input.mouse',
  'input.keyboard',
  'media.control',
  'presentation.control',
  'clipboard.read',
  'clipboard.write',
  'files.read',
  'files.write',
  'apps.launch',
  'windows.control',
  'automation.execute',
  'power.lock',
  'power.sleep',
  'power.restart',
  'power.shutdown',
];

export const DEFAULT_DEVICE_CAPABILITIES: Capability[] = [
  'input.mouse',
  'input.keyboard',
  'media.control',
  'presentation.control',
  'clipboard.read',
  'clipboard.write',
  'apps.launch',
  'windows.control',
  'automation.execute',
  'power.lock',
  'power.sleep',
];

File: crates/remote-protocol/src/capabilities.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub enum Capability {
    #[serde(rename = "input.mouse")]
    InputMouse,
    #[serde(rename = "input.keyboard")]
    InputKeyboard,
    #[serde(rename = "media.control")]
    MediaControl,
    #[serde(rename = "presentation.control")]
    PresentationControl,
    #[serde(rename = "clipboard.read")]
    ClipboardRead,
    #[serde(rename = "clipboard.write")]
    ClipboardWrite,
    #[serde(rename = "files.read")]
    FilesRead,
    #[serde(rename = "files.write")]
    FilesWrite,
    #[serde(rename = "apps.launch")]
    AppsLaunch,
    #[serde(rename = "windows.control")]
    WindowsControl,
    #[serde(rename = "automation.execute")]
    AutomationExecute,
    #[serde(rename = "power.lock")]
    PowerLock,
    #[serde(rename = "power.sleep")]
    PowerSleep,
    #[serde(rename = "power.restart")]
    PowerRestart,
    #[serde(rename = "power.shutdown")]
    PowerShutdown,
}

impl Capability {
    pub fn all() -> Vec<Capability> {
        vec![
            Capability::InputMouse,
            Capability::InputKeyboard,
            Capability::MediaControl,
            Capability::PresentationControl,
            Capability::ClipboardRead,
            Capability::ClipboardWrite,
            Capability::FilesRead,
            Capability::FilesWrite,
            Capability::AppsLaunch,
            Capability::WindowsControl,
            Capability::AutomationExecute,
            Capability::PowerLock,
            Capability::PowerSleep,
            Capability::PowerRestart,
            Capability::PowerShutdown,
        ]
    }

    pub fn default_capabilities() -> Vec<Capability> {
        vec![
            Capability::InputMouse,
            Capability::InputKeyboard,
            Capability::MediaControl,
            Capability::PresentationControl,
            Capability::ClipboardRead,
            Capability::ClipboardWrite,
            Capability::AppsLaunch,
            Capability::WindowsControl,
            Capability::AutomationExecute,
            Capability::PowerLock,
            Capability::PowerSleep,
        ]
    }
}
